windows/CodePushPackage.cs

```diff
@@ -0,0 +1,364 @@
+
+            return await GetPackageFolder(packageHash, false);
+        }
+
+
+        public async Task<StorageFile> GetCurrentPackageBundle(string bundleFileName)
+        {
+            StorageFolder packageFolder = await GetCurrentPackageFolder();
+            if (packageFolder == null)
+            {
+                return null;
+            }
+
+            JObject currentPackage = await GetCurrentPackage();
+            string relativeBundlePath = (string)currentPackage[RELATIVE_BUNDLE_PATH_KEY];
+            if (relativeBundlePath == null)
+            {
+                return await packageFolder.GetFileAsync(bundleFileName);
+            }
+            else
+            {
+                return await packageFolder.GetFileAsync(relativeBundlePath);
+            }
+        }
+
+        public async Task<StorageFolder> GetPackageFolder(string packageHash, bool createIfNotExists)
+        {
+            StorageFolder codePushFolder = await GetCodePushFolder();
+            try
+            {
+                packageHash = shortenPackageHash(packageHash);
+                if (createIfNotExists)
+                {
+                    return await codePushFolder.CreateFolderAsync(packageHash, CreationCollisionOption.OpenIfExists);
+                }
+                else
+                {
+                    return await codePushFolder.GetFolderAsync(packageHash);
+                }
+            }
+            catch (FileNotFoundException)
+            {
+                return null;
+            }
+        }
+
+        public async Task<string> GetCurrentPackageHash()
+        {
+            JObject info = await GetCurrentPackageInfo();
+            string currentPackageShortHash = (string)info[CURRENT_PACKAGE_KEY];
+            if (currentPackageShortHash == null)
+            {
+                return null;
+            }
+
+            JObject currentPackageMetadata = await GetPackage(currentPackageShortHash);
+            if (currentPackageMetadata == null)
+            {
+                return null;
+            }
+            else
+            {
+                return (string)currentPackageMetadata[PACKAGE_HASH_KEY];
+            }
+        }
+
+        public async Task<string> GetPreviousPackageHash()
+        {
+            JObject info = await GetCurrentPackageInfo();
+            string previousPackageShortHash = (string)info[PREVIOUS_PACKAGE_KEY];
+            if (previousPackageShortHash == null)
+            {
+                return null;
+            }
+
+            JObject previousPackageMetadata = await GetPackage(previousPackageShortHash);
+            if (previousPackageMetadata == null)
+            {
+                return null;
+            }
+            else
+            {
+                return (string)previousPackageMetadata[PACKAGE_HASH_KEY];
+            }
+        }
+
+        public async Task<JObject> GetCurrentPackage()
+        {
+            string packageHash = await GetCurrentPackageHash();
```

Comment:
how about a GetPackageFromHash(string packageHash) function that GetCurrentPackage and GetPendingPackage wrap?